### PR TITLE
mongodb: auto generate ObjectId for inserted and saved documents

### PR DESCRIPTION
**Pull request by Morriar**
This PR enables the generation of unique ObjectIds for new JsonObject added to collection.

Example when inserting a document:
~~~nit
var obj = new JsonObject
obj["foo"] = "bar"
print obj.to_json
~~~

Object before insertion:
~~~json
{
   "foo": "bar"
}
~~~

After insertion:
~~~nit
collection.insert(obj)
print obj.to_json
~~~
~~~json
{
   "_id": {"$oid": "5578e5e5f34422650c27af01"},
   "foo": "bar"
}
~~~

### CODE REVIEW

**Comment by privat** [comment]
+1

**Comment by privat** [comment]
After a second read, I'm not sure I understand the difference between the `"$oid"` and the `"_id"` keys in persistent JSON objects.

**Comment by privat** [comment]
Get 1500!

**Comment by BlackMinou** [comment]
+1

**Comment by Morriar** [comment]
This is the representation of the BSON object id used by the C wrapper, we don't have any choice here.

**Comment by privat** [comment]
no documentation clarification to answer to my question https://github.com/privat/nit/pull/1500#issuecomment-111322112

File: lib/mongodb/mongodb.nit [hunk 1]
```diff
@@ -175,6 +173,27 @@ class MongoError
 	redef fun to_s do return "{message} (code: {code})"
 end
 
+# MongoDB Object ID representation.
+private class MongoObjectId
+
+	var native: BSONObjectId
+
+	# The unic ID as an MongoDB Object ID string.
```
Comment (R4PaSs) [inline lib/mongodb/mongodb.nit]: :s/unic/unique

**Comment by R4PaSs** [comment]
:shipit: !

**Comment by Morriar** [comment]
Reroll
* [x] better documentation for ObjectID as asked by @privat
* [x] typo from @R4PaSs

**Pull request merged**